lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -322,7 +290,7 @@ public function getParameters()
      *
      * @return Parameter|null The value of the bound parameter, or NULL if not available.
      */
-    public function getParameter($key)
+    public function getParameter($key): ?Parameter
```

Comment:
```suggestion
    public function getParameter(mixed $key): ?Parameter
```

At least that's what the doc block says. Can we narrow that down to `int|string`?

Reply:
That would make sense 👍

Reply:
Incorporated that in #9777

Reply:
```suggestion
    public function getParameter(int|string $key): ?Parameter
```